### DISCUSSION TITLE
Deployments page 404 if environment name contains a dot

### DISCUSSION THREAD

**Discussion by everpcpc** [community/community#69647] [Actions]
### Select Topic Area

Bug

### Body

For example: 
https://github.com/datafuselabs/bendsql/deployments

But if I open the url
https://github.com/datafuselabs/bendsql/deployments/crates.io
directly, it just returns a 404 Not Found.

**Comment by seyLu** [comment]
#### Hi @everpcpc,

I don't think that is a bug but rather a security feature. I mean you can access files with extensions directly with the url, like `https://<domain>/index.html` is a valid url path.

~~I think what is happening is that, there is some checks on the server on where the request is coming from (or who's making the request). Either that or there are headers attached when you click `deployments/<deployment_name>` on github ui as opposed to just a normal get request.~~

Looking at the network tab:
![image](https://github.com/community/community/assets/98249191/3b88472b-f1ee-40f2-851b-facda0b2926f)
![image](https://github.com/community/community/assets/98249191/c27f34c7-b358-42f5-8e4d-0fcec973d4b5)

This has something to do where the request happened. `same-origin` means the request happened on the same webpage and the same port (eg. clicking `deployment` on github ui) as opposed to `none` which means user-originated operation (eg. pasting url in the address bar)


You can read more on:
- [StackOverflow: Can URLs contain dot](https://stackoverflow.com/questions/6777274/can-urls-contain-dots-in-the-path-part)
- [Web.dev: Fetch metadata](https://web.dev/fetch-metadata/)
- [MDN docs: Sec-Fetch-Site](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-Fetch-Site)

**Comment by everpcpc** [comment]
Hi, @seyLu , I tried deleting the `target=_blank` and click on the deployment link, the response is still 404.
And the `Sec-Fetch-Site` header in request is `same-origin`.

```
:method: GET
:scheme: https
:authority: github.com
:path: /datafuselabs/bendsql/deployments/pypi.org
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Sec-Fetch-Site: same-origin
Cookie: _gh_sess=xxxx
Sec-Fetch-Dest: document
Accept-Language: en-US,en;q=0.9
Sec-Fetch-Mode: navigate
Host: github.com
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.0 Safari/605.1.15
Referer: https://github.com/datafuselabs/bendsql
Accept-Encoding: gzip, deflate, br
Connection: keep-alive
```

```
:status: 404
Content-Type: text/plain; charset=utf-8
Content-Security-Policy: default-src 'none'; base-uri 'self'; connect-src 'self'; form-action 'self'; img-src 'self' data:; script-src 'self'; style-src 'unsafe-inline'
Set-Cookie: xxx
X-XSS-Protection: 0
Content-Encoding: gzip
Referrer-Policy: no-referrer-when-downgrade
Cache-Control: no-cache
Date: Tue, 10 Oct 2023 23:57:17 GMT
X-Content-Type-Options: nosniff
X-Frame-Options: deny
Vary: X-PJAX, X-PJAX-Container, Turbo-Visit, Turbo-Frame, Accept-Encoding, Accept, X-Requested-With
Server: GitHub.com
x-github-request-id: BE51:2801D8:3ABAC0:41BB33:6525E4DC
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
```

**Comment by everpcpc** [comment]
And when clicking on `deployments/<deployment_name>`, GitHub just issues a GraphQL request rather than jumping to the target page.
<img width="496" alt="image" src="https://github.com/community/community/assets/1808802/663e722d-68cc-4b8a-b5eb-c5ff6e425f1b">

**Comment by everpcpc** [comment]
seem resolved
